File: app/src/main/java/com/arjanvlek/oxygenupdater/internal/ThreeTuple.java
package com.arjanvlek.oxygenupdater.internal;

public class ThreeTuple<F, S, T> {

    private F first;
    private S second;
    private T third;

    public static <F, S, T> ThreeTuple<F, S, T> create(F first, S second, T third) {
        ThreeTuple<F, S, T> instance = new ThreeTuple<>();
        instance.first = first;
        instance.second = second;
        instance.third = third;
        return instance;
    }

    public F getFirst() {
        return first;
    }

    public S getSecond() {
        return second;
    }

    public T getThird() {
        return third;
    }
}

File: app/src/main/java/com/arjanvlek/oxygenupdater/internal/ThreeTuple.kt
package com.arjanvlek.oxygenupdater.internal

data class ThreeTuple<F, S, T>(val first: F, val second: S, val third: T) {

    companion object {
        fun <F, S, T> create(first: F, second: S, third: T): ThreeTuple<F, S, T> {
            return ThreeTuple(first, second, third)
        }
    }
}
